clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -6587,3 +6629,26 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
       TranslatorArgs, None));
 }
 
+void SPIRCheck::ConstructJob(Compilation &C, const JobAction &JA,
+                             const InputInfo &Output,
+                             const InputInfoList &Inputs,
+                             const llvm::opt::ArgList &TCArgs,
+                             const char *LinkingOutput) const {
+  // Construct llvm-no-spir-kernel command.
+  assert(isa<SPIRCheckJobAction>(JA) && "Expecting SPIR Check job!");
+
+  // The translator command looks like this:
+  // llvm-no-spir-kernel <file>.bc
+  // Upon success, we just move ahead.  Error means a kernel was found and
```

Comment:
> Error means a kernel was found and

I suggest to treat the checker as the black-box here, as it may evolve - "Error means the check failed"

Reply:
Will adjust.